html/rendering/non-replaced-elements/the-hr-element-0/setting-overflow-visible.html

```diff
@@ -18,27 +27,38 @@
 }
 </style>
 
-<div class=float></div>
-<hr id=test-control>
-<div class=clear></div>
+<div class=wrapper>
+ <div class=float></div>
+ <hr id=test-control>
+ <div class=float></div>
+</div>
 
-<div class=float></div>
-<hr id=test-visible>
-<div class=clear></div>
+<div class=wrapper>
+ <div class=float></div>
+ <hr id=test-visible>
+ <div class=float></div>
+</div>
 
 <script>
-const defaultBodyMargin = 8;
 
 test(() => {
   const hr = document.getElementById('test-control');
-  assert_equals(hr.offsetLeft, 50 + defaultBodyMargin, 'offsetLeft');
-  assert_equals(hr.clientHeight, 50, 'clientHeight');
+  assert_equals(hr.offsetLeft, 50, 'hr.offsetLeft');
+  assert_equals(hr.offsetTop, 0, 'hr.offsetTop');
+  assert_equals(hr.clientHeight, 50, 'hr.clientHeight');
+  const divAfter = hr.nextElementSibling;
+  assert_equals(divAfter.offsetLeft, 0, 'divAfter.offsetLeft');
+  assert_equals(divAfter.offsetTop, 50 + 1 + 1 /* hr border */, 'divAfter.offsetTop');
 }, 'control');
 
 test(() => {
   const hr = document.getElementById('test-visible');
-  assert_equals(hr.offsetLeft, 0 + defaultBodyMargin, 'offsetLeft');
-  assert_equals(hr.clientHeight, 0, 'clientHeight');
+  assert_equals(hr.offsetLeft, 0, 'hr.offsetLeft');
+  assert_equals(hr.offsetTop, 0, 'hr.offsetTop');
+  assert_equals(hr.clientHeight, 0, 'hr.clientHeight');
+  const divAfter = hr.nextElementSibling;
+  assert_equals(divAfter.offsetLeft, 50 + 50, 'divAfter.offsetLeft');
+  assert_equals(divAfter.offsetTop, 1 + 1 /* hr border */, 'divAfter.offsetTop');
```

Comment:
I think the expectation should be "1", not "1+1".

So, you have a 50px tall floated div and a HR whose top border edges are at the same block position. Inside the HR (which does not establish a new BFC) there's another 50px tall float. It's positioned in the content box of the HR, which is just past its top border, i.e. 1px further down. Outside the HR, as the next element, we have what we call |divAfter|, another float. Its top border edge cannot be above the top border edge of any of the other floats in the BFC. The bottommost top border edge is that of the hr::before float, which is at 1px, relatively to .wrapper.

So, what I'm saying is that you shouldn't account for the bottom border of the HR.